### ISSUE TITLE
Supporting Padding in llava processor

### ISSUE THREAD

**Issue by BiEchi** [huggingface/transformers#33175]
### Feature request

https://github.com/haotian-liu/LLaVA uses padding for pre-processing the images by default. Current transformers implementation does not support that.

### Motivation

Request per @NielsRogge at (https://huggingface.co/llava-hf/llava-1.5-7b-hf/discussions/26#66cf46a5a523b74b5f90fa72).

### Your contribution

I successfully reproduced logits after conversion if we add padding in the Transformers library.

**Comment by LysandreJik** [comment]
Thanks for the request @BiEchi! cc @zucchini-nlp

**Comment by MilchstraB** [comment]
Where should I add `do_pad` to achieve the same implementation as original LLaVA.

**Comment by zucchini-nlp** [comment]
cc @NielsRogge as the PR to add `do_pad` is not merged yet

**Comment by NielsRogge** [comment]
The main reason why my PR wasn't merged yet was because
1) there's an open comment to more thoroughly test the `background_color` argument
2) I didn't have equivalence between Pillow and the original implementation for all different values of `background_color` (only got equivalent results for some values).

So @MilchstraB feel free to work further on my PR

**Comment by MilchstraB** [comment]
Thanks for your reply! 

However, if I process images using the `expand2square()` function and pass the processed results into `LlavaProcessor` (`CLIPImageProcessor`), will I get the same result as the original LLaVA?
```python
def expand2square(pil_img, background_color):
    width, height = pil_img.size
    if width == height:
        return pil_img
    elif width > height:
        result = Image.new(pil_img.mode, (width, width), background_color)
        result.paste(pil_img, (0, (width - height) // 2))
        return result
    else:
        result = Image.new(pil_img.mode, (height, height), background_color)
        result.paste(pil_img, ((height - width) // 2, 0))
        return result
    
image = Image.open(<image_path>).convert('RGB')

# processor is a instance of LlavaProcessor
image = expand2square(image, tuple(int(x*255) for x in processor.image_mean))
# Assume there are text prompts
processed_features = processor(images=[image], text=prompt, add_special_tokens=False)
```

**Comment by NielsRogge** [comment]
Yes, you should normally get the same results. 

The problem is that [all image transformations in Transformers](https://github.com/huggingface/transformers/blob/main/src/transformers/image_transforms.py) need to be implemented as functions which take numpy arrays as input and produce other numpy arrays as output, hence one needs to translate the expand2square function to a numpy input, numpy output function with equivalent results.
